test/resource_listings.test.js

```diff
@@ -165,6 +165,48 @@ describe('Listing Resource', function() {
     })
   })
 
+  describe('search', async () => {
+    it('should search listings', async () => {
+      const indexingServerUrl = 'http://hello.world'
+
+      const listings = new Listings({
+        contractService,
+        ipfsService,
+        undefined,
+        indexingServerUrl: indexingServerUrl
+      })
```

Comment:
So this notation is shorthand for passing in a sort of implementation of named parameters. Actually, it's just a destructured javascript object.

This is equivalent to:
```
new Listings({
	contractService: contractService,
	ipfsService: ipfsService,
	undefined: undefined,
	indexingServerUrl: indexingServerUrl
})
```

So in other words you're actually passing in an object property named "undefined" - I'm pretty sure that's not what you intended. If you want to leave a specific named parameter out, you can just omit it from the call.

Another nit is that you can shorten the syntax for `indexingServerUrl: indexingServerUrl`. When you leave out the value of a key/value pair, the value is assumed to be a variable with the same name as the key.

To sum things up, this can all just be written as:

```
const listings = new Listings({
	contractService,
	ipfsService,
	indexingServerUrl
})
```